go/cmd/server/server.go

```diff
@@ -49,21 +52,87 @@ func (s *servingServiceServer) GetOnlineFeatures(ctx context.Context, request *s
 			FeatureNames: &serving.FeatureList{Val: make([]string, 0)},
 		},
 	}
+	// Entities are currently part of the features as a value and the order that we add it to the resp MetaData
+	// Need to figure out a way to map the correct entities to the correct ordering
+	entityValuesMap := make(map[string][]*prototypes.Value, 0)
+	featureNames := make([]string, len(featureVectors))
+	for idx, vector := range featureVectors {
 
-	for _, vector := range featureVectors {
 		resp.Metadata.FeatureNames.Val = append(resp.Metadata.FeatureNames.Val, vector.Name)
-
+		featureNames[idx] = vector.Name
 		values, err := types.ArrowValuesToProtoValues(vector.Values)
 		if err != nil {
 			return nil, err
 		}
-
+		if _, ok := request.Entities[vector.Name]; ok {
+			entityValuesMap[vector.Name] = values
+		}
 		resp.Results = append(resp.Results, &serving.GetOnlineFeaturesResponse_FeatureVector{
 			Values:          values,
 			Statuses:        vector.Statuses,
 			EventTimestamps: vector.Timestamps,
 		})
 	}
 
+	go generateLogs(s, entityValuesMap, featureNames, resp.Results, request.RequestContext)
 	return resp, nil
 }
+
+func generateLogs(s *servingServiceServer, entityMap map[string][]*prototypes.Value, featureNames []string, features []*serving.GetOnlineFeaturesResponse_FeatureVector, requestData map[string]*prototypes.RepeatedValue) error {
+	// Add a log with the request context
+	if len(requestData) > 0 {
+		requestContextLog := logging.Log{
+			RequestContext: requestData,
+		}
+		s.loggingService.EmitLog(&requestContextLog)
```

Comment:
why request data is being emited separately from all the rest?

Reply:
Should this be emitted with each log? Isn't that the same issue as before which is that we only need request context once but it's duplicated for each log?

Reply:
request data is also split by rows, basically has the same structure as entity rows and features

Reply:
Is it ok if I add in request data in a separate pr? Adding this will add a lot more complexity that will require me to update all of the tests again.